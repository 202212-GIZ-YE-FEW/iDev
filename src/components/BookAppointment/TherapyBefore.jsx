import { withTranslation } from "next-i18next";

import RadioGroup from "../ui/radiogroup/RadioGroup";
import RadioInputItem from "../ui/radiogroup/RadioInputItem";

const TherapyBefore = ({ t, name, therapyBefore, onChange }) => {
    return (
        <>
            <RadioGroup title={t("therapyBefore")} asButton={true}>
                <RadioInputItem
                    id='yes'
                    name={name}
                    value='yes'
                    title={t("yes")}
                    checked={therapyBefore === "yes"}
                    onChange={(e) => onChange(e)}
                />
                <RadioInputItem
                    id='no'
                    name={name}
                    value='no'
                    title={t("no")}
                    checked={therapyBefore === "no"}
                    onChange={(e) => onChange(e)}
                />
            </RadioGroup>
        </>
    );
};

export default withTranslation("appointment")(TherapyBefore);
