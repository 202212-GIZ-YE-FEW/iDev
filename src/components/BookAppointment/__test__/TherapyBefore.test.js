import renderer from "react-test-renderer";

import TherapyBefore from "../TherapyBefore";

it("renders correctly", () => {
    const tree = renderer.create(<TherapyBefore />).toJSON();
    expect(tree).toMatchSnapshot();
});
