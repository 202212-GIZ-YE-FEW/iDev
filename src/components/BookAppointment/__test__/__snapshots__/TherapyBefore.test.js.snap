// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <p
    className="mb-4 p-2 text:lg md:text-2xl lg:text-3xl font-normal break-words text-black"
  >
    therapyBefore
  </p>,
  <fieldset
    className="space-y-4"
  >
    <div
      className="text-9xl font-normal text-black"
    >
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="yes"
          onChange={[Function]}
          type="radio"
          value="yes"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="yes"
        >
          yes
        </label>
      </div>
      <div
        className="flex items-center space-y-4"
      >
        <input
          checked={false}
          className="peer hidden"
          id="no"
          onChange={[Function]}
          type="radio"
          value="no"
        />
        <label
          className="w-full flex cursor-pointer rounded-md p-4 text-base md:text-xl lg:text-2xl shadow-md border-[1px] border-light-gray/80 hover:border-gray peer-checked:border-none peer-checked:bg-cyan peer-checked:text-white"
          htmlFor="no"
        >
          no
        </label>
      </div>
    </div>
  </fieldset>,
]
`;
